browser/tor/tor_profile_service.cc

```diff
@@ -51,6 +52,27 @@ void TorProfileService::RegisterLocalStatePrefs(PrefRegistrySimple* registry) {
   }
 }
 
+// static
+std::string TorProfileService::CircuitIsolationKey(const GURL& url) {
+  // https://2019.www.torproject.org/projects/torbrowser/design/#privacy
+  //
+  //    For the purposes of the unlinkability requirements of this
+  //    section as well as the descriptions in the implementation
+  //    section, a URL bar origin means at least the second-level DNS
+  //    name.  For example, for mail.google.com, the origin would be
+  //    google.com.  Implementations MAY, at their option, restrict
+  //    the URL bar origin to be the entire fully qualified domain
+  //    name.
+  //
+  // In particular, we need not isolate by the scheme,
+  // username/password, port, path, or query part of the URL.
+  url::Origin origin = url::Origin::Create(url);
+  std::string domain = net::registry_controlled_domains::GetDomainAndRegistry(
+      origin.host(),
```

Comment:
@riastradh-brave just out of curiosity, what is the difference between this and `url.host()`?